com.unity.perception/Editor/Validation/CharacterTooling.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections.Generic;
+using static UnityEngine.Perception.Content.CharacterValidation;
+using System.Linq;
+
+namespace UnityEngine.Perception.Content
+{
+    public class CharacterTooling : MonoBehaviour
+    {
+        public bool CharacterRequiredBones(Animator animator, out Dictionary<HumanBone, bool> failed)
+        {
+            var result = AvatarRequiredBones(animator);
+            failed = new Dictionary<HumanBone, bool>();
+
+            for (int i = 0; i < result.Count; i++)
+            {
+                var bone = result.ElementAt(i);
+                var boneKey = bone.Key;
+                var boneValue = bone.Value;
+
+                if (boneValue != true)
+                    failed.Add(boneKey, boneValue);
+            }
+
+            if (failed.Count == 0)
+                return true;
+
+            return false;
+        }
+
+        public bool CharacterPoseData(GameObject gameObject, out List<GameObject> failedGameObjects)
+        {
+            failedGameObjects = new List<GameObject>();
+
+            var componentsParent = gameObject.GetComponents<Transform>();
+            var componentsChild = gameObject.GetComponentsInChildren<Transform>();
+
+            for (int p = 0; p < componentsParent.Length; p++)
+            {
+                if (componentsParent[p].GetType() == typeof(Transform))
+                {
+                    var pos = componentsParent[p].transform.position;
+                    var rot = componentsParent[p].transform.rotation.eulerAngles;
+
+                    if (pos == null || rot == null)
```

Comment:
I have a check in here mainly because I found user cases where users would input invalid fbx files and there would be a messed up transform with null values

Reply:
fair enough